static/app/components/events/interfaces/frame/frameVariables.tsx

```diff
@@ -13,6 +13,7 @@ type Props = {
 };
 
 const PYTHON_STRING_REGEX = /^['"](.*)['"]$/;
+const NUMERIC_STRING_REGEX = /^-?\d+(\.\d+)?$/;
```

Comment:
Q: do we ever expect scientific notation `1.234e+56` or hex `0xBADC0FFEE` numerals here?

Reply:
That's a good point! I'll test with the SDK and see how those get formatted

Reply:
It looks like numbers are never stored in that way, e.g. 1e3 gets stored as 1000.0 and 0x10 gets stored as 16